src/libcore/ptr.rs

```diff
@@ -10,9 +10,65 @@
 
 // FIXME: talk about offset, copy_memory, copy_nonoverlapping_memory
 
-//! Raw, unsafe pointers, `*const T`, and `*mut T`.
+//! Manually manage memory through raw pointers.
 //!
 //! *[See also the pointer primitive types](../../std/primitive.pointer.html).*
+//!
+//! # Safety
+//!
+//! Many functions in this module take raw pointers as arguments and dereference
+//! them. For this to be safe, these pointers must be valid. However, because
+//! rust does not yet have a formal memory model, determining whether an
+//! arbitrary pointer is valid for a given operation can be tricky.
+//!
+//! There are two types of operations on memory, reads and writes. A single
+//! pointer can be valid for any combination of these operations. For example, a
+//! pointer is not valid for writes if a `&mut` exists which [refers to the same
+//! memory][aliasing]. The set of operations for which a pointer argument must
+//! be valid is explicitly documented for each function. This is not strictly
+//! necessary for `*const` arguments, as they can only be used for reads and
+//! never for writes.
+//!
+//! Some functions (e.g. [`copy`]) take a single pointer but
+//! operate on many values. In this case, the function will state the size of
+//! the operation for which the pointer must be valid. For example,
+//! `copy::<T>(&src, &mut dst, 3)` requires `dst` to be valid for writes of
+//! `size_of::<T>() * 3` bytes. When the documentation requires that a pointer
+//! be valid for an operation but omits the size of that operation, the size is
+//! implied to be `size_of::<T>()` bytes.
+//!
+//! While we can't yet define whether an arbitrary pointer is a valid one, there
+//! are a few rules regarding validity:
+//!
+//! * The result of casting a reference to a pointer is valid for as long as the
+//!   underlying object is live.
```

Comment:
I'd say something like "a pointer obtained by casting a reference"; had to read this two times to parse it. ;)

Also, this is not true. The pointer obtained from the cast is valid immediately after the cast. However, using the reference again will invalidate the pointer, and other things may also invalidate it. For example:
```rust
fn foo(x: &mut u32) {
  let y = x as *mut;
  *x = 13;
  // Now y is NOT valid any more because `x` got "re-activated"
}
```